omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/DiagnosisController2_2Test.java

```diff
@@ -0,0 +1,262 @@
+		diagnosisSource.add("rank", 1);
+		diagnosisSource.add("encounter", encounter.getUuid());
+		diagnosisSource.add("voided", false);
+		diagnosisSource.add("patient", patient.getUuid());
+		diagnosisSource.add("uuid", "39bf4fbc-fcdb-4a5b-97ea-0d5c4b4315a1");
+		
+		String json = new ObjectMapper().writeValueAsString(diagnosisSource);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		
+		SimpleObject newDiagnosisSource = deserialize(handle(req));
+		
+		Assert.assertNotNull(PropertyUtils.getProperty(newDiagnosisSource, "uuid"));
+		
+		Diagnosis diagnosis = diagnosisService.getDiagnosisByUuid(newDiagnosisSource.get("uuid").toString());
+		
+		LinkedHashMap condition = newDiagnosisSource.get("condition");
+		
+		LinkedHashMap codedOrText = newDiagnosisSource.get("diagnosis");
+		
+		LinkedHashMap conceptName = (LinkedHashMap) codedOrText.get("specificName");
+		
+		LinkedHashMap concept = (LinkedHashMap) codedOrText.get("coded");
+		
+		LinkedHashMap patient = newDiagnosisSource.get("patient");
+		
+		Assert.assertNotNull(diagnosis.getEncounter().toString());
+		Assert.assertEquals(diagnosis.getCondition().getUuid(), condition.get("uuid"));
+		Assert.assertEquals((diagnosis.getCertainty().toString()), newDiagnosisSource.get("certainty"));
+		Assert.assertEquals(diagnosis.getRank(), newDiagnosisSource.get("rank"));
+		Assert.assertEquals(diagnosis.getEncounter().getUuid(), encounter.getUuid());
+		Assert.assertNotNull(newDiagnosisSource.get("encounter"));
+		Assert.assertEquals(diagnosis.getVoided(), newDiagnosisSource.get("voided"));
+		Assert.assertEquals(diagnosis.getDiagnosis().getCoded().getUuid(), concept.get("uuid"));
+		Assert.assertEquals(diagnosis.getDiagnosis().getSpecificName().getUuid(), conceptName.get("uuid"));
+		Assert.assertEquals(patient.get("uuid"), diagnosis.getPatient().getUuid());
+		Assert.assertEquals(originalCount + 1, getAllCount());
+	}
+	
+	@Test
+	public void shouldFetchExistingDiagnosis() throws Exception {
+		
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		SimpleObject result = deserialize(handle(req));
+		
+		Diagnosis diagnosis = diagnosisService.getDiagnosisByUuid(getUuid());
+		
+		Assert.assertEquals(diagnosis.getUuid(), PropertyUtils.getProperty(result, "uuid"));
+	}
+	
+	@Test
+	public void shouldVoidDiagnosis() throws Exception {
+		
+		Diagnosis diagnosis = diagnosisService.getDiagnosisByUuid(getUuid());
+		
+		Assert.assertFalse(diagnosis.isVoided());
+		
+		MockHttpServletRequest req = request(RequestMethod.DELETE, getURI() + "/" + getUuid());
+		req.addParameter("reason", "test");
+		handle(req);
+		
+		diagnosis = diagnosisService.getDiagnosisByUuid(getUuid());
+		Assert.assertTrue(diagnosis.isVoided());
+		Assert.assertEquals("test", diagnosis.getVoidReason());
+		Assert.assertNotNull(diagnosis.getDateVoided());
+		Assert.assertNotNull(diagnosis.getVoidedBy());
```

Comment:
Assert that the voided user is the currently authenticated user.

Reply:
Asking for a bit of clarification. Shouldn't I be asserting if the voided diagnosis is voided by the currently authenticated user?

Reply:
Did you check out how Arthur did it?

Reply:
Yes, I did but He doesn't have an assertion for this.

Reply:
Are you sure you are looking at his full committed work?

Reply:
I rechecked the latest PR and noted the changes done.